### PR TITLE
Add support for --use-application-binary on iOS

### LINKED ISSUE #6283: `flutter run --use-application-binary` doesn't work for iOS
`getApplicationPackageForPlatform` currently doesn't respect the existence of a non-null `applicationBinary`, and instead blindly tries to create an `IOSApp` by calling `new IOSApp.fromCurrentDirectory()`

### PR DESCRIPTION
**Pull request by tvolkert**
Fixes #6283

### CODE REVIEW

**Comment by tvolkert** [comment]
@johnmccutchan @cbracken @danrubel

File: packages/flutter_tools/lib/src/ios/devices.dart [hunk 1]
```diff
@@ -184,17 +184,19 @@ class IOSDevice extends Device {
...
-      printError('Could not build the precompiled application for the device.');
-      diagnoseXcodeBuildFailure(buildResult);
-      printError('');
-      return new LaunchResult.failed();
+    if (!prebuiltApplication) {
+      // TODO(chinmaygarde): Use checked, mainPath, route.
+      // TODO(devoncarew): Handle startPaused, debugPort.
+      printTrace('Building ${app.name} for $id');
+
+      // Step 1: Install the precompiled/DBC application if necessary.
```
Comment (danrubel) [inline packages/flutter_tools/lib/src/ios/devices.dart]: In comment, change Install --> Build ?

File: packages/flutter_tools/lib/src/ios/simulators.dart [hunk 2]
```diff
@@ -415,22 +415,27 @@ class IOSSimulator extends Device {
...
 
     // Prepare launch arguments.
-    List<String> args = <String>[
-      "--flx=${path.absolute(path.join(getBuildDirectory(), 'app.flx'))}",
-      "--dart-main=${path.absolute(mainPath)}",
-      "--packages=${path.absolute('.packages')}",
-    ];
+    List<String> args = <String>[];
+
+    if (!prebuiltApplication)
```
Comment (danrubel) [inline packages/flutter_tools/lib/src/ios/simulators.dart]: nit: curly braces around multi line if statement

**Review by danrubel — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/ios/devices.dart (hunk 1 above)
Comment (tvolkert) [inline packages/flutter_tools/lib/src/ios/devices.dart]: Done

**Comment by johnmccutchan** [comment]
LGTM

**Pull request merged**